api/libproject/cloudsqlproxycontext.rb

```diff
@@ -0,0 +1,28 @@
+require_relative "../../libproject/utils/common"
+
+class CloudSqlProxyContext
+  attr_reader :env_file_path
+
+  def initialize(gcc)
+    @bucket_name = "#{gcc.project}-credentials"
+    @env_file_path = "db/vars.#{gcc.project}.env"
+  end
+
+  def run()
+    common = Common.new
+    common.run_inline %W{
+      docker-compose run --rm api
+        gsutil cp gs://#{@bucket_name}/vars.env #{@env_file_path}
+    }
+    begin
+      common.run_inline %W{docker-compose up -d cloud-sql-proxy}
+      begin
+        yield(@env_file_path)
+      ensure
+        common.run_inline %W{docker-compose down}
```

Comment:
This downs everything, right? Should we do that, or just cloud-sql-proxy?

Reply:
Just `cloud-sql-proxy`, but docker-compose doesn't allow that :(

Reply:
That's pretty bad. It means if I'm running my own api server, etc. while running this command it will die... right?

I'd rather require mysql client installation locally than be stuck with this. Is there a workaround?

Reply:
Can't we just use:

docker-compose stop cloud-sql-proxy

?
https://docs.docker.com/compose/reference/stop/

Reply:
That doesn't remove the stopped containers like `down` does, but I can run `stop` then `rm` to get roughly the same behavior. Your use case is certainly valid, so that does seem safer.

For what it's worth, on future projects I'm going to avoid docker-compose and instead simply create containers using docker commands. That gives much finer-grained control and I'd prefer explicitly creating and destroying resources over the magic of compose.